src/script/run-make.sh

```diff
@@ -53,9 +53,23 @@ function prepare() {
         which_pkg="debianutils"
     fi
 
+    INSTALL_EXTRA_PACKAGES="ccache git $which_pkg clang lvm2"
+    source /etc/os-release
+    if [[ "$ID" == "ubuntu" ]]; then
+        case "$VERSION" in
+            *Jammy*)
+                # when gcc-12 is installed, clang builds try to use it's
+                # version of libstdc++. but nothing depends specifically
+                # on gcc-12's libstdc++ to install it
+                if command -v gcc-12 > /dev/null 2>&1 ; then
+                    INSTALL_EXTRA_PACKAGES+=" libstdc++-12-dev"
```

Comment:
i think a better alternative is to install `g++-12` instead of `gcc-12`.

Reply:
how about this?
```diff
diff --git a/install-deps.sh b/install-deps.sh
index d5eafba48cc..05d5588d43d 100755
--- a/install-deps.sh
+++ b/install-deps.sh
@@ -415,7 +415,7 @@ else
                 ;;
             *Jammy*)
                 [ ! $NO_BOOST_PKGS ] && install_boost_on_ubuntu jammy
-                $SUDO apt-get install -y gcc
+                $SUDO apt-get install -y g++-12
                 ;;
             *)
                 $SUDO apt-get install -y gcc
```

Reply:
in order to avoid running this command on a node which already has g++-12 installed, i'd propose:
```bash
ensure_decent_gcc_on_ubuntu 12 jammy
```